src/python/test/test_dxclient.py

```diff
@@ -1419,6 +1419,15 @@ def test_dx_upload_mult_paths_with_dest(self):
                 listing = run("dx ls /destdir/a").split("\n")
                 self.assertIn(os.path.basename(fd2.name), listing)
 
+    def test_dx_upload_empty_file(self):
+        testdir = tempfile.mkdtemp()
+        os.mkdir(os.path.join(testdir, 'a'))
+        with testutil.TemporaryFile(dir=os.path.join(testdir, "a")) as fd:
+            fd.close()
+            with temporary_project("test_dx_upload_empty_file default", select=True) as upload_empty_file_project_default:
+                run("dx upload {}".format(fd.name))
```

Comment:
I'd prefer to test this by invoking one of the dxpy methods directly (e.g. dxpy.upload_local_file) since the code you added is in the DXFile bindings and not in "dx". It will also eliminate the overhead and annoyances associated with making a subprocess call (get better stack traces, etc.).